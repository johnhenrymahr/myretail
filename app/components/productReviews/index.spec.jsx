import React from 'react'
import { shallow } from 'enzyme'
import ProductReviews from './index'
import renderer from 'react-test-renderer'

describe('product reviews component spec', () => {
  it('renders total reviews link if there are reveiws', () => {
    const props = {
      totalReviews: 5,
      pro: {},
      con: {}
    }
    const wrapper = shallow(<ProductReviews {...props} />)
    expect(wrapper.find('.myRetail-reviews__overall a').length).toEqual(1)
  })
  it('does not render total reviews link when there are no reveiws', () => {
    const props = {
      totalReviews: 0,
      pro: {},
      con: {}
    }
    const wrapper = shallow(<ProductReviews {...props} />)
    expect(wrapper.find('.myRetail-reviews__overall a').length).toEqual(0)
  })
  it('renders correctly as snapshot', () => {
    const props = {
      overallRating: '5',
      totalReviews: 6,
      pro: {
        rating: '4',
        title: 'really good stuff',
        review: 'I really liked it',
        screenName: 'joe good',
        date: '2016-07-06'
      },
      con: {
        rating: '1',
        title: 'really bad stuff',
        review: 'It was horrible',
        screenName: 'sally bad',
        date: '2016-06-16'
      }
    }
    const tree = renderer
      .create(<ProductReviews {...props} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
