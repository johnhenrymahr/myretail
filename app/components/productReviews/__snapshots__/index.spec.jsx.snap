// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product reviews component spec renders correctly as snapshot 1`] = `
<div
  className="myRetail-reviews"
>
  <div
    className="myRetail-reviews__overall"
  >
    <div>
      <span
        className="MyRetail-rating"
      >
        <span
          className="star star--selected"
        >
          ★
        </span>
        <span
          className="star star--selected"
        >
          ★
        </span>
        <span
          className="star star--selected"
        >
          ★
        </span>
        <span
          className="star star--selected"
        >
          ★
        </span>
        <span
          className="star star--selected"
        >
          ★
        </span>
      </span>
       
      <span>
        overall
      </span>
    </div>
    <div>
      <a
        href=""
      >
        view all 
        6
         Reviews
      </a>
    </div>
  </div>
  <div
    className="myRetail-reviews__container myRetail-reviews__container--top row"
  >
    <div
      className="col-xs-6"
    >
      <div
        className="myRetail-reviews__header"
      >
        Pro
      </div>
      <div
        className="myRetail-reviews__subhead"
      >
        most helpfull 4-5 star reveiws
      </div>
    </div>
    <div
      className="col-xs-6"
    >
      <div
        className="myRetail-reviews__header"
      >
        Con
      </div>
      <div
        className="myRetail-reviews__subhead"
      >
        most helpfull 1-2 star reveiws
      </div>
    </div>
  </div>
  <hr />
  <div
    className="myRetail-reviews__container row"
  >
    <div
      className="col-xs-6"
    >
      <div
        className="myRetail-reviews__rating"
      >
        <span
          className="MyRetail-rating"
        >
          <span
            className="star star--selected"
          >
            ★
          </span>
          <span
            className="star star--selected"
          >
            ★
          </span>
          <span
            className="star star--selected"
          >
            ★
          </span>
          <span
            className="star star--selected"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
        </span>
      </div>
      <div
        className="myRetail-reviews__title"
      >
        really good stuff
      </div>
      <div
        className="myRetail-reviews__body"
      >
        I really liked it
      </div>
      <div
        className="myRetail-reviews__footer"
      >
        <span>
          <a
            href=""
          >
            joe good
          </a>
        </span>
        <span>
          July 6, 2016
        </span>
      </div>
    </div>
    <div
      className="col-xs-6"
    >
      <div
        className="myRetail-reviews__rating"
      >
        <span
          className="MyRetail-rating"
        >
          <span
            className="star star--selected"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
          <span
            className="star"
          >
            ★
          </span>
        </span>
      </div>
      <div
        className="myRetail-reviews__title"
      >
        really bad stuff
      </div>
      <div
        className="myRetail-reviews__body"
      >
        It was horrible
      </div>
      <div
        className="myRetail-reviews__footer"
      >
        <span>
          <a
            href=""
          >
            sally bad
          </a>
        </span>
        <span>
          June 16, 2016
        </span>
      </div>
    </div>
  </div>
</div>
`;
